File: GeneratedEntityReaderClasses/Classes/ConfigDamagesByType.cs
namespace GeneratedNoitaClasses;
public class ConfigDamagesByType :  iNoitaType<ConfigDamagesByType>{
	public NoitaFloat melee { get; set; } = new();
	public NoitaFloat projectile { get; set; } = new();
	public NoitaFloat explosion { get; set; } = new();
	public NoitaFloat electricity { get; set; } = new();
	public NoitaFloat fire { get; set; } = new();
	public NoitaFloat drill { get; set; } = new();
	public NoitaFloat slice { get; set; } = new();
	public NoitaFloat ice { get; set; } = new();
	public NoitaFloat healing { get; set; } = new();
	public NoitaFloat physics_hit { get; set; } = new();
	public NoitaFloat radioactive { get; set; } = new();
	public NoitaFloat poison { get; set; } = new();
	public NoitaFloat overeating { get; set; } = new();
	public NoitaFloat curse { get; set; } = new();
	public NoitaFloat holy { get; set; } = new();

    public void Read(NoitaStream s){
		melee.Read(s);
		projectile.Read(s);
		explosion.Read(s);
		electricity.Read(s);
		fire.Read(s);
		drill.Read(s);
		slice.Read(s);
		ice.Read(s);
		healing.Read(s);
		physics_hit.Read(s);
		radioactive.Read(s);
		poison.Read(s);
		overeating.Read(s);
		curse.Read(s);
		holy.Read(s);

    }
    public void Write(NoitaStream s){
		melee.Write(s);
		projectile.Write(s);
		explosion.Write(s);
		electricity.Write(s);
		fire.Write(s);
		drill.Write(s);
		slice.Write(s);
		ice.Write(s);
		healing.Write(s);
		physics_hit.Write(s);
		radioactive.Write(s);
		poison.Write(s);
		overeating.Write(s);
		curse.Write(s);
		holy.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/ConfigDamagesByType.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ConfigDamagesByType:
    melee:NoitaFloat = field(default_factory=NoitaFloat)
    projectile:NoitaFloat = field(default_factory=NoitaFloat)
    explosion:NoitaFloat = field(default_factory=NoitaFloat)
    electricity:NoitaFloat = field(default_factory=NoitaFloat)
    fire:NoitaFloat = field(default_factory=NoitaFloat)
    drill:NoitaFloat = field(default_factory=NoitaFloat)
    slice:NoitaFloat = field(default_factory=NoitaFloat)
    ice:NoitaFloat = field(default_factory=NoitaFloat)
    healing:NoitaFloat = field(default_factory=NoitaFloat)
    physics_hit:NoitaFloat = field(default_factory=NoitaFloat)
    radioactive:NoitaFloat = field(default_factory=NoitaFloat)
    poison:NoitaFloat = field(default_factory=NoitaFloat)
    overeating:NoitaFloat = field(default_factory=NoitaFloat)
    curse:NoitaFloat = field(default_factory=NoitaFloat)
    holy:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.melee.Read(s)
        self.projectile.Read(s)
        self.explosion.Read(s)
        self.electricity.Read(s)
        self.fire.Read(s)
        self.drill.Read(s)
        self.slice.Read(s)
        self.ice.Read(s)
        self.healing.Read(s)
        self.physics_hit.Read(s)
        self.radioactive.Read(s)
        self.poison.Read(s)
        self.overeating.Read(s)
        self.curse.Read(s)
        self.holy.Read(s)
    
    def Write(self, s:NoitaStream):
        self.melee.Write(s)
        self.projectile.Write(s)
        self.explosion.Write(s)
        self.electricity.Write(s)
        self.fire.Write(s)
        self.drill.Write(s)
        self.slice.Write(s)
        self.ice.Write(s)
        self.healing.Write(s)
        self.physics_hit.Write(s)
        self.radioactive.Write(s)
        self.poison.Write(s)
        self.overeating.Write(s)
        self.curse.Write(s)
        self.holy.Write(s)

